test/static/expected_metadata_metadataExtensions.xml

```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0"?>
+<EntityDescriptor xmlns="urn:oasis:names:tc:SAML:2.0:metadata" xmlns:ds="http://www.w3.org/2000/09/xmldsig#" xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" entityID="http://example.serviceprovider.com" ID="d700077e-60ad-49c1-b93a-dd1753528708">
```

Comment:
Ok, I might have mislead because I read what you were doing too quickly; now you have the same namespace listed twice here. The problem is that what you are adding isn't a metadata _extension_, it is just metadata. If you look at the spec on line 385 you'll see that a non-SAML namespace should be used for extensions.

What you're trying to do is identified on lines 432-434 and is a well-known SAML spec, so we should provide some structure for it instead of allowing just any object to be inserted into the metadata (following our existing pattern). Each of the three types of metadata you mention in the description of this PR should have its own set of properties, or take an object, or probably an array of objects, that it duck-typed at run-time and ideally type-checked by TypeScript.

Reply:
I've made a fix